UI

cptKNJO opened the pull request:
This PR adds the following features:
- Show the progress of running backup operations in the status table
- Regularly update the status table every few seconds (configurable via const variable)

(The last commit corresponds to the refetching every 5 seconds. I am combining them because it is a small change, and is useful to test the first feature.)

Displaying both the points here:
![Peek 2024-08-08 23-54](https://github.com/user-attachments/assets/8b50f696-914e-4271-b6c6-f254d7628c1a)

Note: I couldn't test the image backups properly because I couldn't troubleshoot it. Going by the forums, seems like a common issue on my system. (Ubuntu, with dattobd snapshot)

cptKNJO committed:
Style format with prettier

cptKNJO committed:
Update time formatting to match original format

cptKNJO committed:
Add progress display for backup operations

cptKNJO committed:
Add refetch every 5s for status table

uroni commented on urbackupserver/www2/src/features/status/LastBackups.tsx:
```diff
@@ -0,0 +1,115 @@
+    if (result) {
+      clearResultById(id, result.start_type);
+    }
+  }, []);
```
This looks like it might cause (performance) issues?
Is there some other way to solve this, including something like only displaying the message for a certain amount of time and then removing it again?

uroni replied:
We can also adjust what the server returns if you have some recommendations...

cptKNJO replied:
Thank you for pointing this out!

I added a commit which does it in a simpler and more performant way, taking your recommendation:
- keeps a list of backup results
- newest backup results are displayed on the table
- each list of backup results is **cleared every 3s** (from the highest component provider), starting from the oldest backup results, if any

Please let me know if this seems acceptable.

uroni commented:
Overall, looks great!

cptKNJO committed:
Refactor how backup results are cleared

We keep an array of backup results now and only show the status from the latest backup results. Backup results are cleared every 3s, starting from the oldest backup.

cptKNJO committed:
Add styles to status update on table

cptKNJO commented:
I also added a commit to better indicate failure/success status in the table:
![image](https://github.com/user-attachments/assets/ebe0f915-c13c-4c10-97c2-bdd9989fd007)
